Library/Homebrew/keg_relocate.rb

```diff
@@ -70,12 +72,22 @@ def relocate_text_files(old_prefix, new_prefix, old_cellar, new_cellar)
 
   def change_dylib_id(id, file)
     puts "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}" if ARGV.debug?
-    install_name_tool("-id", id, file)
+
+    if ENV["HOMEBREW_RUBY_MACHO"]
+      MachO::Tools.change_dylib_id(file, id)
+    else
+      install_name_tool("-id", id, file)
+    end
   end
```

Comment:
As in defining `change_dylib_id` separately in two modules and including the correct one at the top of `Keg` and other classes that require Mach-O operations?